src/coreclr/runtime.proj

```diff
@@ -5,7 +5,8 @@
   <Target Name="Build">
     <ItemGroup>
       <_CoreClrBuildArg Condition="'$(TargetArchitecture)' != ''" Include="-$(TargetArchitecture)" />
-      <_CoreClrBuildArg Include="$(CMakeArgs)" />
+      <_CoreClrBuildArg Condition="$([MSBuild]::IsOsPlatform(Windows)) and '$(CMakeArgs)' != ''" Include="-cmakeargs &quot;$(CMakeArgs)&quot;" />
+      <_CoreClrBuildArg Condition="!$([MSBuild]::IsOsPlatform(Windows))" Include="$(CMakeArgs)" />
```

Comment:
wouldn't it make sense to modify build-runtime.sh to also take `-cmakeargs` instead of having this difference here?

Reply:
Looks like the script should already handle the -cmakeargs on none Windows platfroms, so not sure why it was done not using it in first place, but maybe the argument in build-commons.sh was added afterwards. I will just move over to using -cmakeargs on both platforms and we see how that behaves on CI.

Reply:
Maybe it had a little different meaning in these scripts, since it was just passed at end, it ended up as unprocessed build args that in turn was only passed to msbuild.sh/msbuild.ps1 in build-runtime.sh/build-runtime.cmd, so previously cmakeargs wouldn't reach into other scripts called from build-runtime.sh/build-runtime.cmd. So then we probably need to pass it using both mechanism, but not sure that will break anything since we will start to pass CMakeArgs from runtime.proj down to build-runtime and then down to commands like build_native, gen-buildsys.cmd.

Reply:
I can do the change in this PR, passing CMakeArgs using both mechanism and see how it works on CI, but I guess we also need a review by someone more familiar with how CMakeArgs is used within current build and if passing it along into gen-buildsys.cmd or build_native could have any side effects.

Reply:
Guess an alternative could be to remove dependency on __UnprocessedBuildArgs to include cmake args and use cmakeargs variable instead, but regardless change, we will end up passing same set of cmakeargs to both msbuild.sh and build_native in build-runtime.sh, so we will need to validate that it won't cause any side effects.

Reply:
Removed dependency on __UnprocessedBuildArgs to be mechanism passing along CMakeArgs to msbuild.sh and msbuild.ps1. Instead we will pass content of __CMakeArgs since values are now passed as part of -cmakeargs inot build-runtime.sh|build-runtime.cmd meaning it won't be part of __UnprocessedBuildArgs.

Reply:
Explicitly passing parsed __CMakeArgs into msbuild.ps1|sh didn't work since it is unknown msbuild arguments.

Reply:
Reverted back to pass -cmakearg down to build-runtime.cmd|sh on both platforms and only include $(CMakeArgs) as and item in _CoreClrBuildArg on none Windows platforms to be consistent with old behaviour.

Reply:
Went back to leave Unix as is and only pass along the CMakeArgs on Windows using -cmakearg parameter in runtime.proj. The current way of passing along CMakeArgs in runtime.proj, will fail on Windows in call to msbuild with errors like -DFEATURE_PERFTRACING_C_LIB=1 is an unknown msbuild switch, so before this PR, getting a value in runtime.proj $(CMakeArgs) wouldn't work on Windows. This PR will then keep behaviour on Unix as is and enable use of runtime.proj $(CMakeArgs) on Windows (by using -cmakearg argument in runime-build.cmd).